yourbench/pipeline/summarization.py

```
@@ -0,0 +1,272 @@
+    try:
+        summ_model_name = list(response_dict.keys())[0]
+        model_raw_summaries: List[str] = response_dict.get(summ_model_name, [])
+
+        if not model_raw_summaries:
+            logger.error("Model '{}' returned no summaries. Check your model configuration.", summ_model_name)
+            return
+
+    except IndexError:
+        logger.error("No valid model keys found in the response dictionary.")
+        return
+
+    if len(model_raw_summaries) != len(documents):
+        logger.warning("Mismatch in number of summaries vs documents. Adjusting list size.")
+        while len(model_raw_summaries) < len(documents):
+            model_raw_summaries.append("")
+        if len(model_raw_summaries) > len(documents):
+            model_raw_summaries = model_raw_summaries[: len(documents)]
+
+    extracted_summaries: List[str] = []
+    for i, raw_resp in enumerate(model_raw_summaries):
+        logger.debug("Parsing doc index {}, raw response length={}", i, len(raw_resp))
+        try:
+            parsed = extract_content_from_xml_tags(raw_resp, "final_summary")
+        except Exception as parse_exc:
+            logger.error("Error parsing doc index {}: {}", i, str(parse_exc))
+            parsed = ""
+
+        parsed_stripped = parsed.strip()
+        if not parsed_stripped:
+            logger.warning("No <final_summary> content found for doc index {}.", i)
+            extracted_summaries.append("No summary available for this document.")
+        else:
+            extracted_summaries.append(parsed_stripped)
+
+    # 5) Add new columns to the dataset
+    try:
+        dataset = dataset.add_column("raw_document_summary", model_raw_summaries)
+    except Exception as e:
+        logger.error("Error adding 'raw_document_summary': {}", str(e))
+
+    try:
+        dataset = dataset.add_column("document_summary", extracted_summaries)
+    except Exception as e:
+        logger.error("Error adding 'document_summary': {}", str(e))
+
+    try:
```

We should consider a better way to handle errors instead of using a broad `try/except`. Catching a wide `Exception` can suppress errors in situations where failure is actually desirable

---

Agreed that a more specific exception management will be good to add in the long term